OpenTOSCA/ui
# Completion refactoring

#### Short Description
Refactoring of the Completion Component (detect CSARs with incomplete requirements and guide through dialog to complete CSAR with valid * templates). Reworked UI with PrimeNG components.
This works for uploading CSARs via the "normal" upload, as well as for marketplace CSARs.








Step 1 - Disclaimer:
![image](https://user-images.githubusercontent.com/18167234/57929950-f9172200-78b4-11e9-8e2c-7c38141354ed.png)


Step 2 - Host Selection:
![image](https://user-images.githubusercontent.com/18167234/57929985-10560f80-78b5-11e9-8ade-4d36c13a8437.png)


Step 3 - Connection Selection:
![image](https://user-images.githubusercontent.com/18167234/57930012-24017600-78b5-11e9-8116-0425a8e846e9.png)


Step 4 - Upload Confirmation:
![image](https://user-images.githubusercontent.com/18167234/57930028-32e82880-78b5-11e9-9051-0266e8b693cb.png)

#### To test this (with docker-compose):
1. Set PUBLIC_HOSTNAME and CONTAINER_REPOSITORY_HOSTNAME in docker-compose override file to your IP address
2. Start ecosystem with docker-compose up
3. Serve UI at port 4300 (or just another port than 4200)
4. Upload 'ConnectsTo.csar', 'DataProvider.csar', 'HostedOn.csar', 'VSphere_Provider.csar' (could not attach it here) to  http://your-ip-address:8091
5. Open UI at http://your-ip-address:4300 
6. Upload incomplete CSAR 'MyTinyToDo_For_Injection.csar' (could not attach it here)
7. Completion Dialog should open now (okay not immediately, this takes a bit:-) )

#### Proposed Changes
  * PrimeNG Components for Completion Dialog UI: Steps and Dropdown
  * Refactoring of Completion Component

**Fixes**: 
* fix catching 406 upload error
* generally, fix completion component not working (with current master version)
